Gems/ROS2/Code/Source/RobotImporter/xacro/XacroUtils.cpp

```diff
@@ -0,0 +1,142 @@
+
+        myProcess.start(program, arguments);
+
+        if (myProcess.waitForFinished() && myProcess.exitStatus() == QProcess::NormalExit && myProcess.exitCode() == 0)
+        {
+            AZ_Printf("ParseXacro", "xacro finished with success");
+            auto output = myProcess.readAllStandardOutput();
+            outcome.m_urdfHandle = UrdfParser::Parse(output.data());
+        }
+        else
+        {
+            AZ_Printf("ParseXacro", "xacro finished with code %d", myProcess.exitCode());
+            auto std = myProcess.readAllStandardOutput();
+            auto cerr = myProcess.readAllStandardError();
+            outcome.m_logStandardOutput = AZStd::string(std.data(), std.size());
+            outcome.m_logErrorOutput = AZStd::string(cerr.data(), cerr.size());
+        }
+        outcome.m_returnCode = myProcess.exitCode();
+        return outcome;
+    }
+
+    AZStd::string WhichXacro()
+    {
+        const QString program{ "which" };
+        QStringList arguments;
+        arguments << kXacroExecutable;
+        QProcess myProcess;
+        myProcess.start(program, arguments);
+        if (myProcess.waitForFinished() && myProcess.exitStatus() == QProcess::NormalExit && myProcess.exitCode() == 0)
+        {
+            auto output = myProcess.readAllStandardOutput().trimmed();
+            return AZStd::string{ output.data(), static_cast<AZStd::string::size_type>(output.size()) };
+        }
+        return AZStd::string{};
+    }
+
+    AZStd::unordered_map<AZStd::string, AZStd::string> GetParameterFromXacroData(AZStd::vector<char> data)
+    {
+        const char* kArgName = "xacro:arg";
+        const char* kName = "name";
+        const char* kDefault = "default";
+
+        AZStd::unordered_map<AZStd::string, AZStd::string> params;
+        AZ::rapidxml::xml_document<char> doc;
+        doc.parse<AZ::rapidxml::parse_full>(data.data());
+        AZ::rapidxml::xml_node<char>* xmlRootNode = doc.first_node("robot");
+        for (AZ::rapidxml::xml_node<>* child_node = xmlRootNode->first_node(); child_node; child_node = child_node->next_sibling())
+        {
+            if (strcmp(kArgName, child_node->name()) == 0)
+            {
+                const auto* attributeName = child_node->first_attribute(kName);
+                const auto* attributeDefault = child_node->first_attribute(kDefault);
+
+                if (attributeName)
+                {
+                    AZStd::string name{ attributeName->value() };
+                    AZStd::string value;
+                    if (attributeDefault)
+                    {
+                        value = AZStd::string{ attributeDefault->value() };
+                    }
+                    params.emplace(AZStd::make_pair(AZStd::move(name), AZStd::move(value)));
+                }
+            }
+        }
+        return params;
+    }
+
+    AZStd::unordered_map<AZStd::string, AZStd::string> GetParameterFromXacroFile(AZStd::string filename)
```

Comment:
nit: Would be nice to pass the filename by `const` ref (or use a `string_view` potentially)